packages/playwright-core/src/server/chromium/crNetworkManager.ts

```diff
@@ -153,19 +153,34 @@ export class CRNetworkManager {
 
   _onAuthRequired(event: Protocol.Fetch.authRequiredPayload) {
     let response: 'Default' | 'CancelAuth' | 'ProvideCredentials' = 'Default';
+    const shouldProvideCredentials = this._shouldProvideCredentials(event.request.url);
     if (this._attemptedAuthentications.has(event.requestId)) {
       response = 'CancelAuth';
-    } else if (this._credentials) {
+    } else if (shouldProvideCredentials) {
       response = 'ProvideCredentials';
       this._attemptedAuthentications.add(event.requestId);
     }
-    const { username, password } = this._credentials || { username: undefined, password: undefined };
+    const { username, password } =  shouldProvideCredentials && this._credentials ? this._credentials : { username: undefined, password: undefined };
```

Comment:
You don't need `&& this._credentials` check here

Reply:
Yeah, I did not have it at first, but I ended up adding it because without it the lint check fails (even though the build work). My understanding was that on that line, this._credentials can be null, so in that case it cannot map the username and password from this._credentials to the const {username, password}.